File: Arrays/kadane(max subarr sum).py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total, maxi = 0, nums[0]
        
        for i in range(len(nums)):
            total += nums[i]
            maxi = max(maxi, total)
            
            if(total < 0): total = 0
                
        return maxi

File: Arrays/kadane(max subarr sum).cpp

//Sol2 Kadane Algorithm | tc:o(N) sc:o(1)
int maxSubArray(vector<int>& nums) {
        int max = nums[0], sum = 0, n = nums.size();
        
        for(int i = 0; i < n; i++){
            sum += nums[i];
            if(sum > max)
                max = sum;
            if(sum < 0)
                sum = 0;
        }
        
        return max;
    }
